qiskit/providers/aer/backends/backend_utils.py

```diff
@@ -77,3 +90,31 @@ def available_devices(controller, devices):
         if result.get('success', False):
             valid_devices.append(device)
     return valid_devices
+
+
+def add_final_save_instruction(qobj, state):
+    """Add final save state instruction to all experiments in a qobj."""
+
+    def save_inst(num_qubits):
+        """Return n-qubit save statevector inst"""
+        return QasmQobjInstruction(
+            name=f"save_{state}",
+            qubits=list(range(num_qubits)),
+            label=f"{state}",
+            snapshot_type="single")
+
+    for exp in qobj.experiments:
+        num_qubits = exp.config.n_qubits
+        exp.instructions.append(save_inst(num_qubits))
+
+    return qobj
```

Comment:
This is to maintain the functionality of these backends with the new controller. The old controllers always added this instruction in the C++ code. This is now done in Python rather than the general controller used for all the backends.

Reply:
It perhaps wouldn't hurt to comment on the reason in the file, since the git log of the file will make it look a little like this behaviour was added in this PR.